src/components/TerraformConfig/TerraformConfig.tsx

```diff
@@ -0,0 +1,136 @@
+import { Alert, Button, Modal, useStyles2 } from '@grafana/ui';
+import { config as runtimeConfig } from '@grafana/runtime';
+import { InstanceContext } from 'contexts/InstanceContext';
+import React, { useContext, useState } from 'react';
+import { Clipboard } from 'components/Clipboard';
+import { css } from '@emotion/css';
+import { GrafanaTheme2 } from '@grafana/data';
+import { TFCheckConfig, TFConfig, TFProbeConfig } from './terraformTypes';
+import { checkToTF, probeToTF, sanitizeName } from './terraformConfigUtils';
+import { checkType } from 'utils';
+import { CheckType } from 'types';
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  modal: css`
+    max-height: 100%;
+  `,
+  clipboard: css`
+    max-height: 500px;
+  `,
+  text: css`
+    max-width: 600px;
+  `,
+});
+
+export const TerraformConfig = () => {
+  const { instance } = useContext(InstanceContext);
+  const [showModal, setShowModal] = useState(false);
+  const [error, setError] = useState<string | undefined>();
+  const [config, setConfig] = useState<TFConfig | undefined>();
+  const styles = useStyles2(getStyles);
+
+  const generateTFConfig = async (): Promise<TFConfig> => {
+    const checks = await instance.api?.listChecks();
+    const probes = await instance.api?.listProbes();
+    if (!checks || !probes) {
+      throw new Error("Couldn't generate TF config");
+    }
+    const checksConfig = checks?.reduce<TFCheckConfig>((acc, check) => {
+      const type = checkType(check.settings);
+      if (type === CheckType.Traceroute) {
+        return acc;
+      }
+      const checkConfig = checkToTF(check);
+      const formattedJob = sanitizeName(check.job);
+      if (!acc[formattedJob]) {
+        acc[formattedJob] = checkConfig;
+      } else {
+        throw new Error(`Cannot generate TF config for checks with duplicate job names: ${check.job}`);
```

Comment:
what's unique is the job / target pair, job names can be duplicated, and in a sense they are expected to be.

Reply:
🤔 I might have to think about this. The problem is that in order to generate TF config, we have to use JSON, and the way the grafana terraform provider is structured expects checks to be identified by a single string key. I don't want to use the target as a json key because I think that will create it's own set of problems, so the job made the most sense. I could probably either generate a UUID, or just append something to the job name to guarantee uniqueness.

Reply:
Just to illustrate the issue, the output config looks like this:
```JSON
    resource: {
      grafana_synthetic_monitoring_check: {
        a_jobname: {
          job: 'a jobname',
          target: 'example.com',
          enabled: true,
          probes: [1],
          labels: {},
          settings: { ping: { ip_version: 'V4', dont_fragment: false } },
        },
          some_other_name: {
          job: 'some other name',
          target: 'example.com',
          enabled: true,
          probes: [1],
          labels: {},
          settings: { ping: { ip_version: 'V4', dont_fragment: false } },
        },
      },
      }
      ```